sdk/data/azcosmos/partition_key.go

```diff
@@ -38,6 +38,14 @@ func NewPartitionKeyNumber(value float64) PartitionKey {
 	}
 }
 
+// NewPartitionKeyNull creates a partition key with a null value.
+func NewPartitionKeyNull() PartitionKey {
```

Comment:
Typically, sentinel-like values are defined as a global var.
```go
var NullPartitionKey PartitionKey = {...}
```

CC @JeffreyRichter

Reply:
It is idiomatic Go to expose sentinel values via a global var (as Joel says) but this can introduce bugs because customers can overwrite the var value. So const is better when possible but it's not possible here.  So, we can go with Joel's suggestion and hope for the best.

Reply:
@karunmotorq Can you follow the suggested change?